server/planning/content_api/events/event.py

```diff
@@ -15,13 +15,32 @@
     ElasticResourceConfig,
     RestEndpointConfig,
 )
+from superdesk.core.resources.service import AsyncResourceService
+
 from content_api import MONGO_PREFIX, ELASTIC_PREFIX
+from planning.output_formatters import JsonEventFormatter
 from planning.content_api.types.events import ContentAPIEventResource
-from superdesk.core.resources.service import AsyncResourceService
 
 
 class ContentAPIEventService(AsyncResourceService[ContentAPIEventResource]):
-    pass
+    """Service for publishing events to the content API"""
+
+    formatter = JsonEventFormatter()
+
+    async def publish_async(self, item, subscribers=None) -> None:
+        """
+        Uses the `JsonEventFormatter` to format the event and publish it to the content API.
+        If the event already exists, it will be updated, otherwise it will be created.
+        """
+
+        formatted_item = await self.formatter._format_item(item)
```

Comment:
It would be beneficial to add error handling and logging to this section to ensure that any unhandled exceptions are properly caught. thoughts ?

Reply:
exceptions are being handled from outside in the consumer so it should be fine.